ops/testnet.py

```diff
@@ -0,0 +1,162 @@
+from os import path, mkdir
+from typing import Tuple, List
+from subprocess import run, check_output, PIPE, STDOUT
+from socket import socket
+from tempfile import gettempdir
+from shutil import rmtree
+
+from config import CONFIG
+from utils import genDockerService
+REPOS = CONFIG["REPOS"]
+
+clientTomlTemplate = """
+[Logging]
+  Disable = false
+  Level = "DEBUG"
+  File = ""
+
+[UpstreamProxy]
+  Type = "none"
+
+[Debug]
+  DisableDecoyTraffic = {}
+  CaseSensitiveUserIdentifiers = false
+  PollingInterval = 1
+
+[NonvotingAuthority]
+    Address = "{}:{}"
+    PublicKey = "{}"
+"""
+
+def generateTestnetConf(ip: str, confDir: str) -> List[str]:
+    """
+    Runs the genconfig tool at a specified directory
+    and returns a list of paths of the generate configs
+    """
+    try: rmtree(confDir)
+    except FileNotFoundError: pass
+    mkdir(confDir)
+    output = run([
+        "genconfig",
+        "-o",
+        confDir,
+        "-n",
+        str(CONFIG["TEST"]["NODES"]),
+        "-a",
+        ip,
+        "-p",
+        str(CONFIG["TEST"]["PROVIDERS"]),
+    ], stdout=PIPE, stderr=STDOUT)
+    return [p.split(" ")[-1] for p in output.stdout.decode().strip().split("\n")]
+
+def getPublicKey(dirPath: str) -> str:
+    """Gets the public key from identity.public.pem file in given directory"""
+    with open(path.join(dirPath, "identity.public.pem"), 'r') as f:
+        return f.read().split("\n")[1] # line index 1
+
+
+def getMixnetPort(path: str) -> str:
+    """Gets mixnet port number from a given katzenpost.toml file"""
```

Comment:
Sure I can add the toml parser. But then pip it is one more additional dependency that needs to be installed and handled. I tried avoiding installing pip packages because then i would have to deal with it for just a toml or yaml parser.

I have no problem using pip, virtualenv, or nix-shells, they all work great. I just seem to have a hard time of making it reliable on other peoples system for just a string parser.

Reply:
By dealing with it i mean that I have to tell users how to install pip or tell them to use their prefer virtual env solution. The docker swarm system I'm using is already causing some friction from what I can see in your other comments and I would prefer to avoid another dependency manager for this PR